components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/ActionExecutorServiceImpl.java

```diff
@@ -67,7 +71,10 @@ public class ActionExecutorServiceImpl implements ActionExecutorService {
     private static final Log LOG = LogFactory.getLog(ActionExecutorServiceImpl.class);
 
     private static final ActionExecutorServiceImpl INSTANCE = new ActionExecutorServiceImpl();
+    private static final ActionExecutionDiagnosticLogger DIAGNOSTIC_LOGGER = new ActionExecutionDiagnosticLogger();
     private final APIClient apiClient;
+    private final ExecutorService executorService = ThreadLocalAwareExecutors.newFixedThreadPool(
+            Runtime.getRuntime().availableProcessors() * 2);
```

Comment:
We can make this Runtime.getRuntime().availableProcessors() * 2 as a static final constant as thread pool size

Reply:
Updated in